go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -318,17 +318,28 @@ func (tp *TablePlan) bindFieldVal(field *querypb.Field, val *sqltypes.Value) (*q
 		return sqltypes.StringBindVariable(valString), nil
 	}
 	if enumValues, ok := tp.EnumValuesMap[field.Name]; ok && !val.IsNull() {
-		// The fact that this fielkd has a EnumValuesMap entry, means we must
+		// The fact that this field has a EnumValuesMap entry, means we must
 		// use the enum's text value as opposed to the enum's numerical value.
 		// Once known use case is with Online DDL, when a column is converted from
 		// ENUM to a VARCHAR/TEXT.
 		enumValue, enumValueOK := enumValues[val.ToString()]
 		if !enumValueOK {
 			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "Invalid enum value: %v for field %s", val, field.Name)
 		}
-		// get the enum text fir this val
+		// get the enum text for this val
 		return sqltypes.StringBindVariable(enumValue), nil
 	}
+	if field.Type == querypb.Type_ENUM {
+		// This is an ENUM w/o a values map, which means that we are most likely using
```

Comment:
you say "most likely": is it possible that we don't get the index here but the value? In that case this would fail, because we would be dealing with strings and not ints, right?

Reply:
Yeah, I think it's possible. That's why if we cannot convert the value to a uint64 then we don't return an error and we let it do what it was doing before the PR, the fallback of `sqltypes.ValueBindVariable(*val)`:
```
                enumIndexVal := sqltypes.MakeTrusted(querypb.Type_UINT64, val.Raw())
 		if enumIndex, err := enumIndexVal.ToUint64(); err == nil {
 			return sqltypes.Uint64BindVariable(enumIndex), nil
 		}
 	}
 	return sqltypes.ValueBindVariable(*val), nil
```